core/modules/replica/HttpModule.cc

```diff
@@ -112,4 +113,33 @@ bool HttpModule::localLoadSecondaryIndex(string const& database) const {
     return false;
 }
 
+
+DatabaseInfo HttpModule::getDatabaseInfo(string const& func) const {
+    debug(func);
+    auto const databaseServices = controller()->serviceProvider()->databaseServices();
+    auto const config = controller()->serviceProvider()->config();
+    string database;
+    if (body().has("database")) {
+        database = body().required<string>("database");
+    } else {
+        if (!body().has("transaction_id")) {
+            throw invalid_argument(
+                    context() + "::" + func + " this service expects either 'database' or "
+                    " 'transaction_id' to be provided to define a scope of the request.");
+        }
+        TransactionId const transactionId = body().required<TransactionId>("transaction_id");
+        debug(func, "transactionId=" + to_string(transactionId));
+        auto const transactionInfo = databaseServices->transaction(transactionId);
+        database = transactionInfo.database;
+    }
+    debug(func, "database=" + database);
+
+    auto const databaseInfo = config->databaseInfo(database);
+    if (databaseInfo.isPublished) {
+        throw HttpError(
+                context() + "::" + func, "database '" + databaseInfo.name + " is already published.");
+    }
```

Comment:
why is it an error to get database info for a published database?

Reply:
The services that are using the method require that the database was still open for ingests (not _published_).